### PR TITLE
8260407: cmp != __null && cmp->Opcode() == Op_CmpL failure with -XX:StressLongCountedLoop=200000000 in lucene

### PR DESCRIPTION
**Pull request by rwestrel**
Long counted loops are transformed in 3 steps:

1- LongCountedLoopNode/LongCountedLoopEndNode are created (similar to
int counted loops)

2- a loop nest is built from the
LongCountedLoopNode/LongCountedLoopEndNode with an inner loop with a
int iv

3- the inner loop is transformed to a CountedLoop

Between 1- and 2-, the LongCountedLoopEndNode is transformed by split
if because a dominated if uses the same condition. Fix simplys applies
existing check for CountedLoopEndNode to LongCountedLoopEndNode.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8260407](https://bugs.openjdk.java.net/browse/JDK-8260407): cmp != __null && cmp->Opcode() == Op_CmpL failure with -XX:StressLongCountedLoop=200000000 in lucene

### Reviewers
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**) ⚠️ Review applies to 8b3312aa7341f57f827f5ca5869e8e1c5d1c4e05
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to 8b3312aa7341f57f827f5ca5869e8e1c5d1c4e05

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2240/head:pull/2240`
`$ git checkout pull/2240`

### CODE REVIEW

File: test/hotspot/jtreg/compiler/longcountedloops/TestLongCountedLoopSplitIf.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2020, Red Hat, Inc. All rights reserved.
```
Comment (TobiHartmann) [inline test/hotspot/jtreg/compiler/longcountedloops/TestLongCountedLoopSplitIf.java:2]: Copyright should be 2021.

**Review by TobiHartmann — APPROVED** [review]
Looks good to me.

**Review by vnkozlov — APPROVED** [review]
Good.

**Comment by rwestrel** [comment]
@TobiHartmann @vnkozlov Thanks for the reviews (I fixed the copyright)

**Pull request closed without merging**